### DISCUSSION TITLE
useContractWrite if ??

### DISCUSSION THREAD

**Discussion by nexos12** [wevm/wagmi#2009] [Q&A]
how to use useContractWrite if the mint function requires both number of mints and it has a price

**Comment by web3dopamine** [comment]
You can you pass `args` to the `useContractWrite` function

the good practice is you first write `usePrepareContractWrite` and set the args there

for more info: https://wagmi.sh/react/prepare-hooks/usePrepareContractWrite

**Reply by dHarshMakwana** [reply]
hey i wanted to use onSuccess for the useContractWrite but its now working its giving me error is there any way to use it or alternative?
